prov/verbs/src/verbs_info.c

```diff
@@ -726,6 +730,72 @@ static int vrb_have_device(void)
 	return ret;
 }
 
+static void vrb_disable_hmem(struct fi_info *info)
+{
+	info->caps &= ~FI_HMEM;
+	info->tx_attr->caps &= ~FI_HMEM;
+	info->rx_attr->caps &= ~FI_HMEM;
+	info->domain_attr->mr_mode &= ~FI_MR_HMEM;
+}
+
+static void vrb_alter_hmem_attrs(struct fi_info *info)
+{
+	static const char *hmem_devices[] = {
+		"mlx4",
+		"mlx5",
+	};
```

Comment:
We can probably just look for 'mlx' to be in the name.  And that check should be first, so we can avoid the extra overhead of opening the file system to read additional data.

Reply:
Sounds good.